### ISSUE TITLE
Add search / filter to make it easier to find tasks

### ISSUE THREAD

**Issue by AnotherKamila** [AnotherKamila/tasksched#27]
Add a search field that filters tasks.

Do not try to re-implement taskwarrior's filtering -- just take the text in the search field and send a request to the server every time. It's not worth it to do it locally.

This needs support in the server -- it should accept the search (e.g. as a query string parameter) and add it to its filter.

**Comment by evilham** [comment]
I _think_ the server doesn't need changes :). If I recall correctly: when I modified the thing, I made the assumption that this would come :-D.
https://github.com/AnotherKamila/tasksched/blob/master/server.js#L21

**Comment by AnotherKamila** [comment]
@evilham yay, you're awesome :D I was too lazy to check when typing that :D great, so it's just a client-side change!

**Comment by evilham** [comment]
BTW, a couple things that came to mind:
* refresh shouldn't lose filters
* maybe we want to allow for multiple saved (locally) saved filters
* there's also the thing of synchronisation (of that hypothetical list) across tabs
* it should be possible to have different filters on different tabs though
* that would mean that any custom filters should be visible on the UI
* maybe it'd be a time to show the name of the current context if one is active?

Basically, it should be easy to know what one is seeing :-D
What do you think? :)

**Comment by AnotherKamila** [comment]
Am I right to think that pretty much all of these requirements are
fulfilled by stuffing the filter into the URL? :-)

**Comment by evilham** [comment]
@AnotherKamila *ping* :-D
